### ISSUE TITLE
Implement IPacketDeserializer interface in the  Rhisis.Network\Packets classes

### ISSUE THREAD

**Issue by tekinomikata** [Eastrall/Rhisis#303]
Maybe not used anymore, but we should consider implementing the `IPacketDeserializer` interface in the  `Rhisis.Network\Packets` classes instead of passing the `INetPacketSteam` through the constructor.

_Originally posted by @Eastrall in https://github.com/Eastrall/Rhisis/pull/301_

**Comment by tekinomikata** [comment]
So you mean removing all the constructors and replacing it with the IPacketDeserializer everywhere?

**Comment by Eastrall** [comment]
Yes, at the begining of the project, all packet were defined as `struct`s. If was used like this in the handlers:
```cs
public void OnCertify(ILoginClient client, INetPacketStream packet)
{
    var certifyPacket = new CertifyPacket(packet);
}
```

Now that we have introduced the `IPacketDeserializer` and the automatic packet reading on handlers, we have something more like:
```cs
public void OnCertify(ILoginClient client, CertifiyPacket packet)
{
}
```
Here, you say the `OnCertify()` method will try to read a `CertifyPacket`. Also, in the `CertifyPacket` class, you can inject whatever service or configuration you want thanks to the shared dependency injector.

So yes, I think it's better to migrate all the `struct`s to `class`es and implement this behavior. I don't think it will change for now.

**Comment by Eastrall** [comment]
Solved in #304
